packages/schematics/src/collection/ng-add/index.ts

```diff
@@ -460,6 +468,61 @@ function dedup(array: any[]): any[] {
   return res;
 }
 
+function insertInToString(originalString: string, pos: number, toAdd: string) {
+  return originalString.slice(0, pos) + toAdd + originalString.slice(pos);
+}
+
+function addNxModule(options: Schema) {
+  return (host: Tree, context: SchematicContext) => {
+    const angularJson = readJsonInTree(host, 'angular.json');
+    const app = angularJson.projects[options.name];
+    const modulePath = path.resolve(
+      '.',
+      getAppModulePath(host, app.architect.build.options.main).slice(1)
+    );
+    let content = fs.readFileSync(modulePath).toString();
```

Comment:
why reading using fs instead of host?

Reply:
if we use host, the whole method will become much simpler because we have a lot of utilities for that already

Reply:
The module file is moved from src to apps/appname/src, So it doesn't appear in the Tree. 

We can use the Tree when we can move directorie